src/python/pants/backend/shell/target_types.py

```diff
@@ -442,6 +461,41 @@ class ShellCommandTarget(Target):
     )
 
 
+class ShellRunInSandboxTarget(Target):
+    alias = "experimental_run_in_sandbox"
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        RunInSandboxRunnableField,
+        RunInSandboxArgumentsField,
+        ShellCommandExecutionDependenciesField,
+        ShellCommandOutputDependenciesField,
+        ShellCommandLogOutputField,
+        ShellCommandOutputsField,
+        RunInSandboxSourcesField,
+        ShellCommandTimeoutField,
+        ShellCommandToolsField,
+        ShellCommandExtraEnvVarsField,
+        EnvironmentField,
+    )
+    help = softwrap(
+        """
+        Execute any runnable target for its side effects.
+
+        Example BUILD file:
+
+            experimental_run_in_sandbox(
+                runnable=":python_source",
+                argv=[""],
+                tools=["tar", "curl", "cat", "bash", "env"],
+                data_dependencies=[":scripts"],
+                outputs=["results/", "logs/my-script.log"],
```

Comment:
This suggests that the only 'observable' side-effects from the runnable are these filesystem changes, right?

A few questions/comments:
1. What happens to stdout and stderr by default? Just captured for logging on failure, etc.?
2. It might be useful to allow 'redirecting' stdout/stderr to a file too. For instance, if the runnable is a little 'glue' shell script, it's often easier to invoke like `./whatever.sh > output.txt`, than support a 'output file' argument and add the appropriate redirection internally. It's a bit easier to do proper output file stuff in a real language (like Python). Thoughts? (Maybe not required for the very first version of this)

Reply:
I noticed that particular issue as well, and was considering adding `capture_stdout` and `capture_stderr` as filename string fields, that would dump the outputs to corresponding files in the root of the sandbox.

I'd definitely be happy to add that as a follow-up, if that suits?

Reply:
> I noticed that particular issue as well, and was considering adding capture_stdout and capture_stderr as filename string fields, that would dump the outputs to corresponding files in the root of the sandbox.

Ah, so something like the following? That seems reasonable (and follow-up is perfectly fine by me)

```python
experimental_run_in_sandbox(
    runnable=[":whatever"],
    ...,
    capture_stdout="path/to/file.txt"
    output_directories=["path/"]
)
```

(Assuming that https://github.com/pantsbuild/pants/pull/17744 applies to this too)